pkg/apis/provisioning/v1alpha5/kubelet_configuration.go

```diff
@@ -22,4 +22,10 @@ type KubeletConfiguration struct {
 	// Note that not all providers may use all addresses.
 	//+optional
 	ClusterDNS []string `json:"clusterDNS,omitempty"`
+	// MaxPods is the maximum number of Pods that can run on the Kubelet. While the value must be
+	// a non-negative integer, some providers might use separate flags to indicate whether a value
+	// will be specified. Therefore, if a negative value is provided, it indicates this field should
+	// not be specified on the underlying kubelet and defaults should be used.
+	//+optional
+	MaxPods int32 `json:"maxPods,omitempty"`
```

Comment:
Can we make this a *int32 so it doesn't default to zero?

Reply:
Absolutely! Blame that on my Go ignorance 😉

Reply:
One thing to keep in mind is that 0 cannot be a valid value. Setting 0 would be the same as omitting the flag, leaving Kubelet to use its own default. This won't be supported by Karpenter, as Karpenter has now knowledge of what that default may be. So technically it should be fine to say that 0 is the same as EKS AMI behavior.